### PR TITLE
Add access token to AppExtension type

### PR DESCRIPTION
**Pull request by korycins**
I want to merge this change because it adds accessToken to AppExtension type.

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

File: saleor/graphql/app/resolvers.py [hunk 1]
```diff
@@ -24,6 +27,19 @@ def resolve_access_token(info, root, **_kwargs):
     return create_access_token_for_app(root, user)
 
 
+def resolve_access_token_for_app_extension(info, root, **_kwargs):
+    user = info.context.user
+    if user.is_anonymous:
+        return None
```
Comment (IKarbowiak) [inline saleor/graphql/app/resolvers.py:33]: You can use here `user_is_valid` method.

**Review by IKarbowiak — APPROVED** [review]

File: saleor/core/jwt.py [hunk 2]
```diff
@@ -179,28 +180,56 @@ def get_user_from_access_payload(payload: dict) -> Optional[User]:
...
         exp_delta=settings.JWT_TTL_APP_ACCESS,
         additional_payload=additional_payload,
     )
     return jwt_encode(payload)
+
+
+def create_access_token_for_app(app: "App", user: "User"):
+    """Create access token for app.
+
+    App can use user jwt token to proceed given operation on the Saleor side.
```
Comment (maarcingebala) [inline saleor/core/jwt.py:211]:
```suggestion
    App can use user's JWT token to proceed given operation in Saleor.
```

File: saleor/graphql/schema.graphql [hunk 3]
```diff
@@ -299,6 +299,8 @@ type AppExtension implements Node {
   type: AppExtensionTypeEnum!
   target: AppExtensionTargetEnum!
   permissions: [Permission!]!
+  accessToken: String
+  canBeUsedByUser: Boolean
```
Comment (maarcingebala) [inline saleor/graphql/schema.graphql:303]: `canBeUsedByUser: Boolean!`

**Comment by patrys** [comment]
Task linked: [SALEOR-4024 Include access_token  and can_be_used_by user to AppExtension](https://app.clickup.com/t/2549495/SALEOR-4024)

File: saleor/graphql/schema.graphql (hunk 3 above)
Comment (korycins) [reply saleor/graphql/schema.graphql:303]: My bad, `canBeUsedByUser` will be added in separate PR as I have to (probably) add some performance improvments for these fields. Will revert this field, but thanks for catch

**Review by maarcingebala — APPROVED** [review]

**Pull request merged**